src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageReadmeControl.xaml.cs

```diff
@@ -22,16 +25,47 @@ public partial class PackageReadmeControl : UserControl
 #pragma warning disable CS0618 // Type or member is obsolete
         private IMarkdownPreview _markdownPreview;
 #pragma warning restore CS0618 // Type or member is obsolete
-
+        private bool _isInitializing = false;
         public PackageReadmeControl()
         {
             InitializeComponent();
-            _markdownPreview = MarkdownPreviewSingleton.GetInstance();
-            descriptionMarkdownPreview.Content = _markdownPreview.VisualElement;
         }
 
         public ReadmePreviewViewModel ReadmeViewModel { get => (ReadmePreviewViewModel)DataContext; }
 
+        private async Task InitializeAsync()
+        {
+            if (_markdownPreview is null && !_isInitializing)
+            {
+                _isInitializing = true;
+                await TaskScheduler.Default;
+                IEditorOptionsFactoryService editorOptionsFactoryService;
+                try
+                {
+                    var componentModel = await AsyncServiceProvider.GlobalProvider.GetComponentModelAsync();
+                    editorOptionsFactoryService = componentModel?.GetService<IEditorOptionsFactoryService>();
+                }
+                finally
+                {
+                    await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
By the time this resumes, your UI could have been closed and disposed:

To demo this this add this after the `TaskScheduler.Default`:

``` C#
await Task.Delay(15000)
```

Then close the UI before that 15 seconds times out, you will see that you crash Visual Studio when you resume.

What you should be doing is passing a CancellationToken that represents your "UI control/pages lifetime" and then you will be guaranteed to not resume on the UI thread should that cancellation token be disposed on the UI thread and cancelled in Dispose.

Reply:
Updated so it's no longer async